[Fire Mage] Enhancements and Additions for Hot Streak

Sharrq opened the pull request:
Hot Streak has been expanded to once again check for direct damage crits while Hot Streak is active (but now also disregards crits that happen when Pyromaniac procs and immediately re-activated Hot Streak).

Also added checks to make sure each use of hot streak has a hard cast ability like Fireball, Scorch, or Pyroblast (only if you have a bracer proc) immediately before it to increase the chance of getting a Heating Up or Hot Streak ... unless Combustion is active.

kfinch commented on src/Parser/Mage/Fire/Modules/Features/HotStreak.js:
```diff
@@ -30,115 +32,182 @@ class HotStreak extends Analyzer {
+      });
+      when(this.wastedCritsThresholds)
+        .addSuggestion((suggest, actual, recommended) => {
+          return suggest(<span>You crit with {formatNumber(this.wastedCrits)} direct damage abilities while <SpellLink id={SPELLS.HOT_STREAK.id} /> was active. This is a waste since those crits could have contibuted towards your next Hot Streak. Try to use your procs as soon as possible to avoid this.</span>)
```
`<Wrapper>` instead of `<span>` please.

kfinch commented on src/Parser/Mage/Fire/Modules/Features/HotStreak.js:
```diff
@@ -30,115 +32,182 @@ class HotStreak extends Analyzer {
+
+  get wastedCritsThresholds() {
+    return {
+      actual: this.wastedCrits,
```
Don't make suggestions based on absolute number... this looks like an appropriate time to use wasted crits per minute.

Sharrq commented:
I did test it, ill try to test it a little more in depth but on the 3-4 fights i tried it on it worked as it should. 

I should have time tonight so ill make the other suggested changes and get a little more testing in.

kfinch commented:
I think I get largely what is going on, fix the `<span>`s and I can merge

Sharrq commented:
I actually found another issue with it, so I'm trying to fix that first

Sharrq commented:
I fixed the issue I found with Scorch Damage Events happening at the exact same timestamp as Hot Streak being removed

Also added a check for Firestarter.
